src/tests/test_pubkey.cpp

```diff
@@ -630,6 +630,31 @@ std::vector<Test::Result> PK_Key_Generation_Test::run() {
                            !public_key->supports_operation(Botan::PublicKeyOperation::KeyAgreement));
          }
 
+         // Test that the raw public key can be encoded. RSA is excused, as it
+         // does not have an obvious raw encoding.
+         if(public_key->algo_name() != "RSA") {
```

Comment:
I'd rephrase this test slightly.

1) Always invoke `raw_public_key_bits` (for example, here we never call it on RSA, so if it simply crashed or threw some unexpected exception, we wouldn't notice).

2) The test may throw an exception. If the exception is not `Not_Implemented`, failure. If the exception is `Not_Implemented`, failure unless `algo_name() == "RSA"`.

3) Test of test as is.